modules/compiler/spirv-ll/source/context.cpp

```diff
@@ -146,8 +146,15 @@ cargo::expected<spirv_ll::Module, spirv_ll::Error> spirv_ll::Context::translate(
   // have been generated
   llvm::SmallVector<OpIRLocTy, 8> Phis;
 
+  llvm::Error error = llvm::Error::success();
+  // We need to check this error - even though it's obviously success - because
+  // we're going to re-assign it and that destroys it. *All* Error objects must
+  // be checked before they're destroyed.
```

Comment:
We may be able to use `std::optional<llvm::Error>` to avoid having to do special weird things needing a big comment here? If we do that, I think it can also be local to the loop as it was before.

Reply:
Great idea. That works, thank you!